src/main/scala-3/com/github/takezoe/scaladoc/EmbedScaladocAnnotationPlugin.scala

```diff
@@ -0,0 +1,42 @@
+package com.github.takezoe.scaladoc
+
+import dotty.tools.dotc.plugins.{StandardPlugin, PluginPhase}
+import dotty.tools.dotc.ast.tpd.*
+import dotty.tools.dotc.core.Contexts.Context
+import dotty.tools.dotc.core.Constants.Constant
+import dotty.tools.dotc.core.Symbols.requiredClass
+import dotty.tools.dotc.core.StdNames.nme
+import dotty.tools.dotc.core.Comments.{Comment, docCtx}
+import dotty.tools.dotc.core.Annotations.Annotation
+
+import dotty.tools.dotc.util.Spans
+
+class EmbedScaladocAnnotationPlugin extends StandardPlugin:
+  override val name: String = "EmbedScaladocAnnotation"
+  override val description: String = "Embeds Scaladoc comments as runtime annotations"
+
+  def init(options: List[String]): List[PluginPhase] =
+    List(EmbedScaladocAnnotationPhase())
+
+class EmbedScaladocAnnotationPhase extends PluginPhase:
+  val phaseName = "EmbedScaladocAnnotation"
+  override val runsAfter = Set("typer")
+  override val runsBefore = Set("checkUnusedPostTyper")
+
+  private def addScaladocAnnotation(tree: DefTree)(using ctx: Context): Tree =
+    ctx.docCtx.getOrElse:
+      throw RuntimeException("Internal error: DocCtx could not be found and documentations are unavailable. Please report this issue to the maintainers.")
+    .docstring(tree.symbol).foreach:
+      case Comment(span, comment, _, _, _) =>
+        val annotationSymbol = requiredClass("com.github.takezoe.scaladoc.Scaladoc")
+        tree.symbol.addAnnotation(Annotation(annotationSymbol, List(Literal(Constant(comment))), span))
+    tree
+
+  override def transformValDef(tree: ValDef)(using Context): Tree =
```

Comment:
If you never produce new trees but rather only visit existing one and add some additional annotations then you can use prepareValDef instead. This variant of visitor methods returns unit